### PR TITLE
fix(app): shore up edge cases of current command tracking, use new commands endpoint link metadata

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Fixes a bug where the current command item disappears periodically and miscalculates when LPC has
been run prior to starting the protocol. Also handles fast scrolling that spans multiple windows.

Note: in order to test this protocol the backend changes in #9400 will need to be on your robot.

Closes https://github.com/Opentrons/opentrons/issues/9379

# Changelog

# Review requests

- play around with the run log after running LPC. Make sure that the current command highlights as expected and the scroll behavior is consistent.

# Risk assessment

### CODE REVIEW

File: app/src/assets/localization/en/protocol_info.json [hunk 1]
```diff
@@ -53,6 +53,6 @@
   "labware_positon_check_complete_toast_with_offsets": "Labware Position Check complete. {{count}} Labware Offset created.",
   "labware_positon_check_complete_toast_with_offsets_plural": "Labware Position Check complete. {{count}} Labware Offsets created.",
   "protocol_loading": "Opening Protocol On {{robot_name}}",
-  "protocol_finishing": "Closing Protocol On {{robot_name}}",
+  "protocol_finishing": "Finishing Protocol On {{robot_name}}",
```
Comment (b-cooper) [inline app/src/assets/localization/en/protocol_info.json:56]: this text was in accurate/confusing, so I opportunistically fixed it

File: app/src/organisms/ProtocolSetup/RunSetupCard/hooks/__tests__/useModuleMatchResults.test.tsx [hunk 2]
```diff
@@ -47,7 +47,7 @@ describe('useModuleMatchResults', () => {
   beforeEach(() => {
     store.dispatch = jest.fn()
 
-    when(mockGetConnectedRobot).mockReturnValue(mockConnectedRobot)
+    mockGetConnectedRobot.mockReturnValue(mockConnectedRobot)
```
Comment (b-cooper) [inline app/src/organisms/ProtocolSetup/RunSetupCard/hooks/__tests__/useModuleMatchResults.test.tsx:50]: Jest was failing for me locally, but it turned out that my version of jest-when was out of sync. This is another reason to pin our dependencies to a version.

File: app/src/organisms/RunDetails/CommandItem.tsx [hunk 3]
```diff
@@ -106,7 +112,7 @@ export function CommandItemComponent(
     analysisCommand?.commandType === 'pause' ||
     runCommandSummary?.commandType === 'pause'
   const backgroundColor =
-    commandStatus === 'queued' && hasBeenRun
+    commandStatus === 'queued' && isMostRecentCommand
```
Comment (shlokamin) [inline app/src/organisms/RunDetails/CommandItem.tsx]: do we need this?

File: app/src/organisms/RunDetails/CommandList.tsx [hunk 4]
```diff
@@ -142,19 +157,49 @@ export function CommandList(): JSX.Element | null {
...
+        currentCommandCreatedAt != null &&
+        new Date(currentCommandCreatedAt) > runStartDateTime
+      ) {
+        // then we know that the run has diverged from the analysis expectation and
+        // that this protocol is non-deterministic
+        setIsDeterministic(false)
+      }
+    }
+  }
+  const isCurrentCommandInFinalWindow =
```
Comment (shlokamin) [inline app/src/organisms/RunDetails/CommandList.tsx:199]: let's comment the edge case this is solving for

**Review by shlokamin — APPROVED** [review]

**Pull request merged**